x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlQueryResponse.java

```diff
@@ -346,18 +348,48 @@ public EsqlResponse responseInternal() {
 
     public static class Profile implements Writeable, ChunkedToXContentObject {
```

Comment:
NIT: I wonder if we could make this a record?
This way we could avoid custom equals/hashCode/toString.
Merge could be updated to return a new instance instead of mutating this one